astore/rpc/astore.proto

```diff
@@ -133,13 +133,23 @@ message NoteResponse {
   repeated Artifact artifact = 1;
 }
 
+message DeleteRequest {
+  string id = 1; //SID or UID (will be interpreted to which based on length)
+}
+
+message DeleteResponse {
+  int32 code = 1; //Valid values are like fs errors, either 0 1 or 2;
```

Comment:
Just return an empty message, no need to define code and message.
gRPC will carry both an error string and error code implicitly for each rpc call.

Just return an error from your server handler, and the client side will receive the same error. 
To specify an error code, you can use [status.Errorf here](https://godoc.org/google.golang.org/grpc/status#Errorf)
If you don't specify an error code explicitly, some default is used (APPLICATION... something something? not sure).

Reply:
Actually, maybe return the list of things that were actually deleted? Up to you.

Reply:
I think that returning a list of ID's is more idiomatic, especially since deleting an SID would result in a bunch of uid's being deleted also.

Reply:
Don't feel strongly. Let's start implementing it, and decide later :)? The goal of returning a list here though is to show the user what was deleted. The uids won't be human friendly/helpful.

Having subsequent lookups for each UID would fail, as the artifact has now been removed from the repository? Well, depends on how we implement the removal, the UID could stick around, just not tied to a path. But if we do provide an API to get all the info out of an UID, it's certainly arguable if it should/should not return results that have been deleted.